### ISSUE TITLE
Add -SolutionDirectory and -PackageDirectory options to nuget.exe pack

### ISSUE THREAD

**Issue by xwlantian** [NuGet/Home#7163]
## Details about Problem
The [GetSolutionDir](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/Common/ProjectHelper.cs#L47) method used by `nuget pack` could return wrong `SolutionDir` if there is a "unrelated" `.sln` file in the project folder or in the folder structure "before" the real `.sln` file. It causes `NU5012` error unable to find the nupkg file because the wrong `SolutionDir` value.

We may need a `-SolutionDirectory` option like in `nuget restore` command to specify `SolutionDirectory` explicity.

**Comment by jainaashish** [comment]
We haven't seen a real customer scenario where there is a `wrong` sln file right next to your project file and `real` sln is some where else... it seems odd. But I'll keep it around and listen to more feedback.

**Comment by xwlantian** [comment]
It's not uncommon that multiple solutions reference the same project but the solution files are in different folder.

**Comment by Stevie-O** [comment]
While trying to migrate one of our internal projects to be referenced using NuGet (currently, the project is referenced via svn:externals and git submodules), we got bitten by this limitation.  Specifically, for historical reasons, the solution file in question is *not* found in a direct ancestor directory, but rather a subdirectory of a common ancestor.

I've made a preliminary patch that adds `-SolutionDirectory` and `-PackagesDirectory` to the `nuget pack` command.  If PackagesDirectory is specified, it is used directly; else if SolutionDirectory is specified, it uses that to locate the packages directory; otherwise, it falls back to the existing code.

There are five things preventing me from creating a PR outright:
1. *Translations*.  Since this involves new text (in `-help` output), it needs to be translated.  I copied the text from the `nuget restore` versions; however, the SolutionDirectory text says "not applicable for a solution file".  Somebody who knows the relevant languages will need to edit the text to remove that addendum.
2. *Tests*.  While I have a rough idea of how the tests should go, I don't understand the testing infrastructure well enough to create an automated test.
3. *PackTask*.  This looks like something used for MSBuild integration.  I have no idea how to begin testing the associated functionality for such a thing.
4. *Branch*.  I'm not sure what the best branch is to base my changes on.
5. *Inelegance*.  While the changes I made to the core code are fairly elegant (IMO), the code populating `PackArgs` is a bit clunky.  I couldn't figure out the most appropriate place to force relative-to-absolute path conversion, and the core `ProjectFactory` class requires absolute paths for correct functioning.
